pkg/cmd/get/token/exec.go

```diff
@@ -0,0 +1,184 @@
+	"open-cluster-management.io/clusteradm/pkg/cmd/init/scenario"
+	"open-cluster-management.io/clusteradm/pkg/config"
+	"open-cluster-management.io/clusteradm/pkg/helpers"
+	"open-cluster-management.io/clusteradm/pkg/helpers/apply"
+
+	"github.com/spf13/cobra"
+
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
+	o.values = Values{
+		Hub: Hub{
+			TokenID:     helpers.RandStringRunes_az09(6),
+			TokenSecret: helpers.RandStringRunes_az09(16),
+		},
+	}
+	return nil
+}
+
+func (o *Options) validate() error {
+	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	apiExtensionsClient, err := apiextensionsclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+	installed, err := helpers.IsClusterManagerInstalled(apiExtensionsClient)
+	if err != nil {
+		return err
+	}
+	if !installed {
+		return fmt.Errorf("this is not a hub")
+	}
+	return err
+}
+
+func (o *Options) run() error {
+	output := make([]string, 0)
+	reader := scenario.GetScenarioResourcesReader()
+
+	kubeClient, err := o.ClusteradmFlags.KubectlFactory.KubernetesClientSet()
+	if err != nil {
+		return err
+	}
+	dynamicClient, err := o.ClusteradmFlags.KubectlFactory.DynamicClient()
+	if err != nil {
+		return err
+	}
+
+	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	apiExtensionsClient, err := apiextensionsclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	clientHolder := resourceapply.NewClientHolder().
+		WithAPIExtensionsClient(apiExtensionsClient).
+		WithKubernetes(kubeClient).
+		WithDynamicClient(dynamicClient)
+
+	token, err := getToken(kubeClient)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		files := []string{
+			"init/namespace.yaml",
+		}
+		if o.useBootstrapToken {
+			files = append(files,
+				"init/bootstrap-token-secret.yaml",
+				"init/bootstrap_cluster_role.yaml",
+				"init/bootstrap_cluster_role_binding.yaml",
+			)
+		} else {
+			files = append(files,
+				"init/bootstrap_sa.yaml",
+				"init/bootstrap_cluster_role.yaml",
+				"init/bootstrap_sa_cluster_role_binding.yaml",
+			)
+		}
+		out, err := apply.ApplyDirectly(clientHolder, reader, o.values, o.ClusteradmFlags.DryRun, "", files...)
+		if err != nil {
+			return err
+		}
+		output = append(output, out...)
+		if !o.ClusteradmFlags.DryRun {
+			//Make sure that the sa token is ready
+			if !o.useBootstrapToken {
+				b := retry.DefaultBackoff
+				b.Duration = 100 * time.Millisecond
+				_, err = waitForBootstrapSecret(kubeClient, b)
+				if err != nil {
+					return err
+				}
+			}
+			token, err = getToken(kubeClient)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		//Update the cluster role
+		files := []string{
```

Comment:
This is the issue that happened today.
I requested the token once, it created the sa and role/binding. Then we changed the role because we were missing something the "create". Requesting again the token, didn't updated the role and so wasn't working. I added the 'else' so at least the role is updated every time if needed.